### PR TITLE
Custom configuration for Satisfaction75

### PR DESCRIPTION
**Pull request by awkannan**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -15,5 +15,7 @@
     "npm-debug.log.*": true,
     "test/**/__snapshots__": true,
     "yarn.lock": true
-  }
+  },
+  "editor.tabSize": 2,
```
Comment (awkannan) [inline .vscode/settings.json]: These agree with our eslint rules

File: package.json [hunk 2]
```diff
@@ -183,16 +184,18 @@
     "history": "^4.7.2",
     "lodash.debounce": "^4.0.8",
     "lottie-web": "^5.3.4",
-    "react": "^16.5.0",
-    "react-dom": "^16.5.0",
+    "react": "^16.8.0",
```
Comment (awkannan) [inline package.json]: let's keep up to date w/ dependencies (also hooks)

File: package.json [hunk 3]
```diff
@@ -183,16 +184,18 @@
...
+    "react-dom": "^16.8.0",
     "react-hot-loader": "^4.3.6",
     "react-redux": "^5.0.6",
     "react-router": "^4.3.1",
     "react-router-dom": "^4.3.1",
     "react-router-redux": "^5.0.0-alpha.6",
+    "react-select": "^2.4.1",
```
Comment (awkannan) [inline package.json]: 🙏 Hoping this is ok

File: package.json [hunk 4]
```diff
@@ -183,16 +184,18 @@
...
     "redux": "^3.7.2",
     "redux-thunk": "^2.2.0",
-    "source-map-support": "^0.5.3"
+    "source-map-support": "^0.5.3",
+    "styled-components": "^4.1.3"
```
Comment (awkannan) [inline package.json]: 🙏 Also hoping this is ok

Comment (awkannan) [reply package.json]: (if this is ok, then I will start moving things over to styled-components, starting with the Key component)

File: app/utils/device-meta.js [hunk 5]
```diff
@@ -71,7 +70,9 @@ export type DeviceMeta = {
   layout: string,
   matrixLayout: string,
   lights: boolean,
-  overrideMatrixIndexing?: boolean
+  overrideMatrixIndexing?: boolean,
+  customConfig?: any,
+  vendorProductId?: number
```
Comment (awkannan) [inline app/utils/device-meta.js]: this should be non-optional once everything moves to a keyboard folder, but for now it's optional.

File: app/utils/device-meta.js [hunk 6]
```diff
@@ -71,7 +70,9 @@ export type DeviceMeta = {
   layout: string,
   matrixLayout: string,
   lights: boolean,
-  overrideMatrixIndexing?: boolean
+  overrideMatrixIndexing?: boolean,
+  customConfig?: any,
```
Comment (awkannan) [inline app/utils/device-meta.js]: not sure what type this should be.

It's a dict mapping a string to a React component that takes in a `KeyboardAPI` object in as a prop.

File: app/keyboards/standards.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+// TODO: link to KLE for standards?
```
Comment (awkannan) [inline app/keyboards/standards.js]: This is supposed to be a spot for standard layouts, so they aren't made a bunch of times.

File: package.json [hunk 8]
```diff
@@ -115,6 +115,7 @@
     "babel-plugin-add-module-exports": "^0.2.1",
     "babel-plugin-dev-expression": "^0.2.1",
     "babel-plugin-flow-runtime": "^0.15.0",
+    "babel-plugin-styled-components": "^1.10.0",
```
Comment (awkannan) [inline package.json]: this enables the `css` prop on all components, via styled components

See this: https://medium.com/styled-components/announcing-native-support-for-the-css-prop-in-styled-components-245ca5252feb

File: app/utils/device-meta.js [hunk 9]
```diff
@@ -86,161 +87,165 @@ export type CompiledDeviceMeta = DeviceMeta & {
...
+  0x65826080: {
     name: 'BIOI G60',
     layout: LAYOUT_G60,
     matrixLayout: MATRIX_G60,
-    lights: false,
+    lights: false
   }
 };
 
+export const DEVICE_META_MAP: DeviceMetaMap = {
```
Comment (olivia) [inline app/utils/device-meta.js]: You can write this as 
```
export const DEVICE_META_MAP: DeviceMetaMap = KEYBOARDS.reduce((accum, deviceMeta) => {.......}, LEGACY_DEVICE_META_MAP);

File: package.json (hunk 4 above)
Comment (olivia) [reply package.json]: 👍

File: package.json (hunk 3 above)
Comment (olivia) [reply package.json]: 👍

File: app/keyboards/satisfaction75/encoder-mode-toggle.js [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+  enabledModes: number,
+  onChange: (value: number) => void
+};
+
+export class EncoderModeToggle extends Component<Props> {
+  handleInputChange = event => {
+    const {enabledModes, onChange} = this.props;
+    const {target} = event;
+    const {checked: value, name} = target;
+
```
Comment (olivia) [inline app/keyboards/satisfaction75/encoder-mode-toggle.js]: For clarity do something like this:
```
const flagBit = 1 << MODES[name];
const newEnabledModes = value ? enabledModes | flagBit : enableModes & ~flagBit;
onChange(newEnabledModes)
```

File: app/keyboards/satisfaction75/encoder-mode-toggle.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+type Props = {
+  enabledModes: number,
+  onChange: (value: number) => void
+};
+
+export class EncoderModeToggle extends Component<Props> {
+  handleInputChange = event => {
+    const {enabledModes, onChange} = this.props;
+    const {target} = event;
```
Comment (olivia) [inline app/keyboards/satisfaction75/encoder-mode-toggle.js]: Just extract checked and name directly from event `const {target: {checked, name}} = event`

File: app/components/Home.js [hunk 12]
```diff
@@ -406,8 +406,11 @@ export default class Home extends React.Component<Props, State> {
     } else if (selectedTitle === Title.LIGHTING && api && false) {
       return <div />;
     } else if (selectedTitle === Title.DEBUG) {
-      return <DebugMenu api={this.getAPI(selectedKeyboard)} />;
+      return <DebugMenu api={api} />;
     }
+    const keyboardMeta = getKeyboardFromDevice(selectedKeyboard);
```
Comment (olivia) [inline app/components/Home.js]: Move this up to where the constants are defined and add another if block like this:
```
const meta = getKeyboardFromDevice(selectedKeyboard);
const CustomMenu = meta.customMenus && meta.customMenus.find(({title}) => title === selectedTitle);
if (CustomMenu) {
 return <CustomMenu api={api} />;
} else if (selectedTitle === Title.KEYS) {
....
```

File: app/keyboards/satisfaction75/config.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+	{ K000,  KC_NO, K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  KC_NO,  KC_NO }, \
+	{ K100,  K101,  K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  KC_NO,  K115 }, \
+	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  KC_NO,  K215 }, \
+	{ K300,  K301,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311,  K312,  KC_NO, KC_NO,  K315 }, \
+	{ K400,  K401,  K402,  K403,  K404,  K405,  K406,  K407,  K408,  K409,  K410,  K411,  KC_NO, K413,  KC_NO,  K415 }, \
+	{ K500,  K501,  K502,  KC_NO, KC_NO, K505,  KC_NO, KC_NO, KC_NO, K509,  K510,  K511,  K512,  K513,  KC_NO,  K515 } \
+}
+`;
+
+const CONFIG = {
```
Comment (olivia) [inline app/keyboards/satisfaction75/config.js]: Rename to `DEVICE_META`

File: app/keyboards/satisfaction75/config.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
+import {SatisfactionMenu} from './satisfaction-menu';
+
+export const LAYOUT_SATISFACTION75 = `[{c:"#09b084",a:7},"0",{x:0.5,c:"#cccccc"},"1","2","3","4",{x:0.25,c:"#09b084"},"5","6","7","8",{x:0.25,c:"#cccccc"},"9","10","11","12"],
```
Comment (olivia) [inline app/keyboards/satisfaction75/config.js]: Call it `LAYOUT`

File: app/keyboards/satisfaction75/config.js [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+[{y:0.25,c:"#09b084"},"13",{c:"#cccccc"},"14","15","16","17","18","19","20","21","22","23","24","25",{c:"#09b084",w:2},"26",{x:0.5},"27"],
+[{w:1.5},"28",{c:"#cccccc"},"29","30","31","32","33","34","35","36","37","38","39","40",{c:"#09b084",w:1.5},"41",{x:0.5},"42"],
+[{w:1.75},"43",{c:"#cccccc"},"44","45","46","47","48","49","50","51","52","53","54",{c:"#09b084",w:2.25},"55",{x:0.5},"56"],
+[{w:2.25},"57",{c:"#cccccc"},"58","59","60","61","62","63","64","65","66","67",{c:"#09b084",w:1.75},"68",{x:1.5},"70"],
+[{y:-0.75,x:14.25},"69"],
+[{y:-0.25,w:1.25},"71",{w:1.25},"72",{w:1.25},"73",{c:"#cccccc",w:6.25},"74",{c:"#09b084"},"75","76","77"],
+[{y:-0.75,x:13.25},"78","79","80"]`;
+
+export const MATRIX_SATISFACTION75 = `
```
Comment (olivia) [inline app/keyboards/satisfaction75/config.js]: Rename to `MATRIX`

**Comment by olivia** [comment]
LGTM!

**Pull request merged**